xapian-core/tests/harness/backendmanager_remotetcp.cc

```diff
@@ -79,6 +79,9 @@ struct pid_fd {
 
 static pid_fd pid_to_fd[16];
 
```

Comment:
any other way to avoid global variable?

Reply:
It is at least `static`, so it's only global for this one source file.

It needs to be accessible from the `SIGCHLD` handler and from the code which launches new `xapian-tcpsrv` processes, and the only argument that a signal handler gets passed is the signal number, so we can't arrange to pass in a pointer to a structure.

We could have a function which returned a pointer to a structure which was static within that function, but that doesn't seem to really make things cleaner, it just adds a wrapper layer which makes things less clear if anything.

I'd say that when you're dealing with signal handlers and the like, the code is inevitably low level system programming, and a global variable isn't such a bad thing when what's being represented is (as here) information which is naturally a singleton (it wouldn't make sense to have two different `pid_to_fd` structures).